### PR TITLE
Use gradle plugin auto install in onboarding docs

### PR DESCRIPTION
**Pull request by adinauer**
Use the gradle plugins auto install feature we added recently and bump versions of gradle plugin as well as Java SDK.

### CODE REVIEW

**Comment by priscilawebdev** [comment]
the changes look good to me but this will conflict with the PR https://github.com/getsentry/sentry/pull/54675

**Comment by adinauer** [comment]
@priscilawebdev we can hold this PR until the other one is merged. Is there an ETA for it?
I marked this PR as draft for now.

> Btw, shall we also use the sources/fetched version in the links or it shall remain hard-coded?

I'd say let's use latest version for links as well, presuming you mean the central.sonatype.com links I changed in this PR.

File: static/app/gettingStartedDocs/java/spring-boot.tsx
Comment (lbloder) [inline static/app/gettingStartedDocs/java/spring-boot.tsx:1]: The Position of the information about the maven plugin for `Source Context` is somewhat confusing.
Maybe have a `Source Context` Section with the maven plugin and a `Logback` Section ?

File: static/app/gettingStartedDocs/java/java.tsx
Comment (lbloder) [inline static/app/gettingStartedDocs/java/java.tsx:1]: Not actually part of the changes in this PR, but the code in the `Measure Performance` section is not correctly formatted.

**Review by lbloder — APPROVED** [review]
Actual changes LGTM.
Left some comments about minor issues

File: static/app/gettingStartedDocs/java/log4j2.tsx
Comment (adinauer) [reply static/app/gettingStartedDocs/java/log4j2.tsx:65]: Ah good catch these no longer seem to work. Maybe @ArthurKnaus has info whether this is already on Tele teams TODO list?

File: static/app/gettingStartedDocs/java/spring-boot.tsx
Comment (adinauer) [reply static/app/gettingStartedDocs/java/spring-boot.tsx:1]: Not really sure whether this should rather just be linked in "Next Steps" as it's more of an advanced topic and onboarding docs should focus on basics. wdyt?

With SAGP we should be installing our logback integration automatically anyways.

File: static/app/gettingStartedDocs/java/java.tsx
Comment (adinauer) [reply static/app/gettingStartedDocs/java/java.tsx:1]: We were thinking about removing this here as well. I don't think other platforms show how to manually instrument performance in onboarding docs.

**Comment by adinauer** [comment]
Keeping this a draft for now since #54675 was reverted due to CORS issues.

File: static/app/gettingStartedDocs/java/log4j2.tsx
Comment (ArthurKnaus) [reply static/app/gettingStartedDocs/java/log4j2.tsx:65]: thx for the input! I have created a task for this in this epic https://github.com/getsentry/sentry/issues/53859 👍

Comment (adinauer) [reply static/app/gettingStartedDocs/java/log4j2.tsx:65]: @ArthurKnaus thanks! Afair certain strings (like `___ORG_SLUG___`) were replaced automatically. Not 100% sure this worked in onboarding docs. Maybe it's not needed anymore once we change to the other kind of auth tokens for uploading source context (blanking on the proper name here).

**Comment by matejminar** [comment]
@adinauer this is now shipped in production https://github.com/getsentry/sentry/issues/53921 👍

File: static/app/gettingStartedDocs/java/spring-boot.tsx
Comment (lbloder) [reply static/app/gettingStartedDocs/java/spring-boot.tsx:1]: Sounds good to me 👍

**Comment by adinauer** [comment]
@ArthurKnaus and @lbloder can you please give this another pass?

File: static/app/gettingStartedDocs/java/java.tsx
Comment (lbloder) [inline static/app/gettingStartedDocs/java/java.tsx:1]: Should we remove the "manual installation" part for gradle here, and only show the usage of the gradle plugin? Like we do for the other integrations

File: static/app/gettingStartedDocs/java/log4j2.tsx
Comment (lbloder) [inline static/app/gettingStartedDocs/java/log4j2.tsx:50]: Relevant for all integrations:
(m) Maybe add a quick (1 sentence) explainer that for gradle you just need to add the plugin and sentry and integrations will be automatically installed by the plugin.

**Review by ArthurKnaus — APPROVED** [review]
Nice work! 👍

File: static/app/gettingStartedDocs/java/java.tsx
Comment (adinauer) [inline static/app/gettingStartedDocs/java/java.tsx:46]: @ArthurKnaus can you please take a look at this link (it's duplicated across all platforms). I'm unsure what this should look like.

Comment (ArthurKnaus) [reply static/app/gettingStartedDocs/java/java.tsx:46]: Is there any reason why we need to use this redirect link?
If not I would have just redirected without leaving the app:
```
<Link to="/settings/auth-tokens/" />
```

**Pull request merged**
